pkg/engine/executor/sortmerge.go

```diff
@@ -0,0 +1,202 @@
+package executor
```

Comment:
nit: I don't think anything in this file needs to be publicly exported as part of the executor API, does it?

Reply:
No, it's not needed. I followed the example of the other implementations. We can do a follow up PR to make them non-exported.